db/crud.go

```diff
@@ -3404,6 +3404,46 @@ func (c *DatabaseCollection) checkForUpgrade(ctx context.Context, key string, un
 	return doc, rawDocument
 }
 
+// legacyRevToHybridLogicalVector will take a legacy revID and convert it to a HybridLogicalVector, used for when a
+// document doesn't have hlv defined (it is a legacy pre upgraded doc).
+func legacyRevToHybridLogicalVector(ctx context.Context, docID, revID string) (hlv *HybridLogicalVector, missing []string, err error) {
+	version, err := LegacyRevToRevTreeEncodedVersion(revID)
+	if err != nil {
+		base.WarnfCtx(ctx, "error parse legacy revID to version for doc %s: %v", base.UD(docID), err)
+		missing = append(missing, revID)
+		return
+	}
+	hlv = NewHybridLogicalVector()
+	err = hlv.AddVersion(version)
+	if err != nil {
+		base.WarnfCtx(ctx, "error add version to hlv for doc %s: %v", base.UD(docID), err)
+		missing = append(missing, revID)
+		return
+	}
+	return hlv, missing, nil
```

Comment:
I was thinking to avoid erroring at a lower level so that we can

```suggestion
	if err != nil {
		return nil, fmt.Errorf("error adding version to hlv for doc %s: %v", base.UD(docID), err)
	}
    return hlv, nil
```